### PR TITLE
Remove libprotobuf dep

### PR DESCRIPTION
**Pull request by jdye64**

### CODE REVIEW

**Comment by jdye64** [comment]
Do not merge until https://github.com/apache/arrow-datafusion-python/pull/527 is merged and this PR is updated first.

File: continuous_integration/environment-3.10-dev.yaml [hunk 1]
```diff
@@ -3,15 +3,15 @@ channels:
...
+# - c-compiler
 - dask>=2022.3.0
 - fastapi>=0.92.0
 - fugue>=0.7.3
 - httpx>=0.24.1
 - intake>=0.6.0
 - jsonschema
 - lightgbm
-- maturin>=1.1,<1.2
+- maturin>=1.3.1
```
Comment (charlesbluca) [inline continuous_integration/environment-3.10-dev.yaml:14]: There are a couple more files we should update this dep in:

- pyproject.toml
- docker/conda.txt
- docker/main.dockerfile
- docs/requirements-docs.txt

Comment (jdye64) [reply continuous_integration/environment-3.10-dev.yaml:14]: Noted and done. Will push with next commit pointing to the merged ADP PR

**Comment by jdye64** [comment]
rerun tests

**Comment by jdye64** [comment]
@charlesbluca Any chance you could take a look at this today?

File: continuous_integration/gpuci/environment-3.10.yaml [hunk 2]
```diff
@@ -6,7 +6,7 @@ channels:
 - nvidia
 - nodefaults
 dependencies:
-- c-compiler
```
Comment (charlesbluca) [inline continuous_integration/gpuci/environment-3.10.yaml:9]: Based on the conda recipe, seems like we still need a C compiler for compilation - do we need to modify this pinning in some way?

File: continuous_integration/gpuci/environment-3.10.yaml [hunk 3]
```diff
@@ -15,7 +15,7 @@ dependencies:
 - intake>=0.6.0
 - jsonschema
 - lightgbm
-- maturin>=1.1,<1.2
```
Comment (charlesbluca) [inline continuous_integration/gpuci/environment-3.10.yaml:18]: Just out of interest, what are the specific features we need from the newer version of maturin? Also might make sense to pin this a little more tightly if the specific version we need is 1.3.1:


```suggestion
- maturin>=1.3.1,<1.4
```

File: continuous_integration/recipe/conda_build_config.yaml [hunk 4]
```diff
@@ -1,4 +1,4 @@
 rust_compiler_version:
-    - 1.69
+    - 1.73
 maturin:
     - 1.1
```
Comment (charlesbluca) [inline continuous_integration/recipe/conda_build_config.yaml:4]: Should change this to match whatever pinned maturin version we converge on; assuming for now that's 1.3.1:


```suggestion
    - 1.3.1
```

**Review by charlesbluca — COMMENTED** [review]
Thanks for this work @jdye64, this looks good so far. Looks like there's still some bits to the wheel release workflow that need to be modified to remove protoc installation, mind if I push some commits to get that working?

**Review by charlesbluca — APPROVED** [review]

**Pull request merged**